### DISCUSSION TITLE
wsl: Unknown key 'wsl2.defaultVhdSize'

### DISCUSSION THREAD

**Discussion by taylorsilva** [microsoft/WSL#12168] [Q&A] [answered]
Starting two weeks ago (first week of October) I started getting this error every time I start up WSL (Ubuntu 22.04):

```
wsl: Unknown key 'wsl2.defaultVhdSize' in C:\Users\tsilva\.wslconfig:5
```

My `.wslconfig` looks like this:
```
[wsl2]
# Limit disk size to 256B
defaultVhdSize=256GB
```

From what I can tell this is still a valid value based on the docs here: https://learn.microsoft.com/en-us/windows/wsl/wsl-config

Just to confirm that my WSL didn't somehow flip to WSL 1 (which I never even used on this computer):

```pwsh
PS C:\Users\tsilva> wsl --list --verbose
  NAME            STATE           VERSION
* Ubuntu-22.04    Running         2
```

To be clear, my WSL is working fine otherwise. This message pops up at the start but everything works fine. I'm just curious _why_ I'd be getting this error and am wondering if anyone else started getting this error too.

**Comment by OneBlue** [comment, marked as answer]
@taylorsilva : You're probably running an old version of WSL that doesn't have that setting. Running wsl --update should fix this.
